Automattic/wp-calypso
# Pattern Library: Pre-populate react-query cache for individual categories after fetching all patterns

Related to https://github.com/Automattic/dotcom-forge/issues/6532

## Proposed Changes

Data fetching for the pattern library works like this:
- Patterns are always fetched from the `/ptk/patterns` API endpoint
- When the user requests an individual category, we query the API endpoint for patterns from that category
- When the user submits a global search, we query the API endpoint for _all_ patterns (and then we filter them client-side)

> [!IMPORTANT]
> There's a gotcha to the `usePatterns` query for _all_ patterns, which is that we currently submit it instantly when the home page loads, we don't wait for the user to submit a search (although this is the conceptual intention). #89478 brings back the intended behavior.

After we've fetched all patterns, we don't technically need to run new queries to fetch patterns from individual categories. We already have all the data necessary to display those patterns.

This PR takes the first step to avoid those unnecessary requests by prepopulating the react-query cache for individual category queries with the data from the _all_ patterns request.

In other words, after the user has submitted a search, navigating to a new category will be instant – all patterns will already be loaded.

## Testing Instructions



1. Navigate to `/patterns`
2. Wait a second or two
3. Click on the `About` category
4. Ensure that the pattern previews are there instantly (there is no short moment when the list of patterns is empty)
